### ISSUE TITLE
Aborts due to issue in node-sql-parser

### ISSUE THREAD

**Issue by TonyGravagno** [araera111/mysql-to-zod#5]
Ref https://github.com/taozhi8833998/node-sql-parser/issues/778#issuecomment-1627368852

I don't know if that issue will affect many uses of mysql-to-zod. I'm just noting the relation here.

**Comment by TonyGravagno** [comment]
The issue with node-sql-parser was closed as fixed. [According to the author](https://github.com/taozhi8833998/node-sql-parser/pull/1488#issuecomment-1629224706) (@taozhi8833998) :

> The new version `4.8.0` will be released this week after some other issues get fixed.

The current version is 4.6.6. I will reload this project and start the init from scratch when nsp 4.8 is available. He did understandably ask me which SQL revealed the issue. This code doesn't show SQL information as it's being processed so at this moment I don't know what I could show him if the issue persists. Please enhance MtZ to show the table names or some other data as they are processed. I'll be happy if you can tell me where to add a console.log 😄. Thanks!

**Comment by araera111** [comment]
The part of my library that uses node-sql-parser is the createSchemaFile function in createSchemeaFile.ts.
It parses the show create table statement in MySQL and converts the result to zod.

```ts
export const createSchemaFile = (
  tableDefinition: string[], // index 0 is table name, 1 is tableDefinition
  options: MysqlToZodOption
) => {
  const parser = new Parser();
  const [tableName, tableDefinitionString] = tableDefinition;
  if (isNil(tableName) || isNil(tableDefinitionString)) return left("createSchemeFileError");

//example add console.log(tableDefinitionString); 

  const ast = parser.astify(tableDefinitionString);
  if (Array.isArray(ast) || !isCreate(ast)) return left("createSchemeFileError");

  const columns = columnsSchema
    .array()
    .parse(
      ast.create_definitions
        ? .map((x: any) => convertToColumn(x))
        .flatMap((x: any) => (isNil(x) ? [] : x))
    );
  const schema = createSchema(tableName, columns, options);
  return schema; }
}
```

The tableDefinitionString is obtained with the getTableDefinition function in dbManipulateFunctions.ts.
It connects directly to mysql and calls ````show create table ? ```` (? = table name). = table name).
You can retrieve the create table statement by running console.log in this area.

Alternatively, you can directly retrieve the data from MySQL.
You can use adminer, phpmyadmin, or directly type [show create table test(dbname)] into MySQL to see the results.
![image](https://github.com/araera111/mysql-to-zod/assets/63596736/6056087d-6583-40c3-8297-8cfeeaaff868)

You can give it to node-sql-parser issue.
(I also had a parse fail once, so I submitted it to the issue to get it fixed.)

I hope this helps you.

I am Japanese and this text is written using translation software. I apologize if it is difficult to understand or if I am rude.

**Comment by taozhi8833998** [comment]
```sql
show create table todo; show tables
```
the above SQLs are supported now. If I missed anything, please let me know.

**Comment by TonyGravagno** [comment]
わかりました。:smile:
ありがとう !

**Comment by taozhi8833998** [comment]
@TonyGravagno I have published the node-sql-parser `4.8.0` version, please upgrade to the latest version, and try again.

**Comment by araera111** [comment]
I have updated. This issue is closed.
